app/src/main/kotlin/com/wire/android/ui/home/conversations/edit/EditMessageMenuItems.kt

```diff
@@ -125,70 +124,117 @@ fun EditMessageMenuItems(
-            if (isGenericAsset) add { OpenAssetExternallyOption(onOpenAssetClick) }
-            if (isEditable && !isEphemeral) {
-                add {
-                    MenuBottomSheetItem(
-                        icon = {
-                            MenuItemIcon(
-                                id = R.drawable.ic_edit,
-                                contentDescription = stringResource(R.string.content_description_edit_the_message)
+                    ReplyMessageOption(onReplyItemClick)
+                    if (isAssetMessage) add { DownloadAssetExternallyOption(onDownloadAssetClick) }
+                    if (isGenericAsset) add { OpenAssetExternallyOption(onOpenAssetClick) }
+                    if (isEditable) {
+                        add {
+                            MenuBottomSheetItem(
+                                icon = {
+                                    MenuItemIcon(
+                                        id = R.drawable.ic_edit,
+                                        contentDescription = stringResource(R.string.content_description_edit_the_message)
+                                    )
+                                },
+                                title = stringResource(R.string.label_edit),
+                                onItemClick = onEditItemClick
                             )
-                        },
-                        title = stringResource(R.string.label_edit),
-                        onItemClick = onEditItemClick
-                    )
+                        }
+                    }
+                    if (isAssetMessage) {
+                        add {
+                            MenuBottomSheetItem(
+                                icon = {
+                                    MenuItemIcon(
+                                        id = R.drawable.ic_share_file,
+                                        contentDescription = stringResource(R.string.content_description_share_the_file),
+                                    )
+                                },
+                                title = stringResource(R.string.label_share),
+                                onItemClick = onShareAsset
+                            )
+                        }
+                    }
+                add {
+                    CompositionLocalProvider(LocalContentColor provides MaterialTheme.colorScheme.error) {
+                        MenuBottomSheetItem(
+                            icon = {
+                                MenuItemIcon(
+                                    id = R.drawable.ic_delete,
+                                    contentDescription = stringResource(R.string.content_description_delete_the_message),
+                                )
+                            },
+                            title = stringResource(R.string.label_delete),
+                            onItemClick = onDeleteItemClick
+                        )
+                    }
```

Comment:
Can we move this MenuBottomSheetItem to its own file, aka `DeleteMessageMenuOption`?